File: src/api/test_metrics.js
const request = require('supertest');
const app = require('../server');

describe('POST /api/metrics', () => {
    it('should submit gameplay metrics successfully', async () => {
        const response = await request(app)
            .post('/api/metrics')
            .send({ userId: 'user123', performanceData: { score: 100, level: 5 } });
        expect(response.status).toBe(201);
        expect(response.body.message).toBe('Metrics submitted successfully.');
    });

    it('should return error for missing userId', async () => {
        const response = await request(app)
            .post('/api/metrics')
            .send({ performanceData: { score: 100, level: 5 } });
        expect(response.status).toBe(400);
        expect(response.body.message).toBe('User ID and performance data are required.');
    });
});

File: src/api/test_metrics.py
import pytest
from fastapi.testclient import TestClient
from .metrics import app

client = TestClient(app)

def test_submit_performance_metrics():
    response = client.post('/api/performance-metrics', json={'metric_name': 'CPU Usage', 'value': 75.5, 'timestamp': '2023-10-01T12:00:00Z'})
    assert response.status_code == 200
    assert response.json() == {'message': 'Metrics submitted successfully!'}


def test_get_performance_metrics():
    client.post('/api/performance-metrics', json={'metric_name': 'Memory Usage', 'value': 65.2, 'timestamp': '2023-10-01T12:01:00Z'})
    response = client.get('/api/performance-metrics')
    assert response.status_code == 200
    assert len(response.json()) > 0